.github/workflows/prerelease.yaml

```diff
@@ -0,0 +1,63 @@
+on:
+  pull_request:
+    branches:    
+      - main
+    types:
+      - edited
+      - opened
+      - synchronize
+      - reopened
+
+name: release-please
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v5
+        id: configure-changelog
+        with:
+          result-encoding: string
+          script: |
+            const changelogTypes = [
+              {type: "feat", section: "Features", hidden: false},
+              {type: "chore", section: "Misc", hidden: false},
+              {type: "fix", section: "BugFixes", hidden: false},
+            ]
+
+            return JSON.stringify(changelogTypes)
+      # See https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow
+      # For why we need to generate a token and not use the default
+      - name: Generate token
+        id: generate_token
+        uses: chanzuckerberg/github-app-token@v1.1.4
+        with:
+          app_id: ${{ secrets.CZI_RELEASE_PLEASE_APP_ID }}
+          private_key: ${{ secrets.CZI_RELEASE_PLEASE_PK }}
+
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.17
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
+      
+      - name: Generate PreRelease Version
+        id: generate_tag
+        run: |
+          VERSION=v$(cat version.txt)-$(git rev-parse --short HEAD)
+          echo "::set-output name=tag-name::$VERSION"
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
+
+      - name: Prerelease
+        uses: goreleaser/goreleaser-action@v2
+        with:
+          version: latest
+          args: release --rm-dist -f .goreleaser.prerelease.yml
+        env:
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
+          GORELEASER_CURRENT_TAG: ${{ steps.generate_tag.outputs.tag-name}}
+        if: ${{ github.event_name == 'release' && github.event.release.prerelease }}
```

Comment:
this block should live int he `release.yml` file and we can get rid of this file. 
then, for these steps, do them when we do not release (in that other action file)

Reply:
On it